awx/main/dispatch/worker/callback.py

```diff
@@ -39,14 +73,9 @@ def perform_work(self, body):
             if not any([key in body for key in event_map]):
                 raise Exception('Payload does not have a job identifier')
 
-            def _save_event_data():
-                for key, cls in event_map.items():
-                    if key in body:
-                        cls.create_from_data(**body)
-
             job_identifier = 'unknown job'
             job_key = 'unknown'
-            for key in event_map.keys():
+            for key, cls in event_map.items():
```

Comment:
We've actually got to track a dictionary of buffers *indexed by* the event type.  That way (over a quarter second), we get:

`JobEvent, JobEvent, ProjectUpdateEvent, InventoryUpdateEvent, JobEvent`.

When it's time to flush, we run:

```python
JobEvent.objects.bulk_create([JE, JE JE])
ProjectUpdateEvent.objects.bulk_create([PUE])
InventoryUpdateEvent.objects.bulk_create([IUE])
```